packages/jest/src/executors/jest/jest.impl.ts

```diff
@@ -150,24 +156,50 @@ export async function batchJest(
   overrides: JestExecutorOptions,
   context: ExecutorContext
 ): Promise<Record<string, { success: boolean; terminalOutput: string }>> {
-  const configPaths = taskGraph.roots.map((root) =>
-    path.resolve(context.root, inputs[root].jestConfig)
-  );
+  let configPaths: string[] = [];
+  let selectedProjects: string[] = [];
+  for (const task of taskGraph.roots) {
+    let configPath = path.resolve(context.root, inputs[task].jestConfig);
+    configPaths.push(configPath);
+
+    /* The display name in the jest.config.js is the correct project name jest
+     * uses to determine projects. It is usually the same as the Nx projectName
+     * but it can be changed. The safest method is to extract the displayName
+     * from the config file, but skip the project if it does not exist. */
+    const displayNameValueRegex = new RegExp(
+      /(['"]+.*['"])(?<=displayName+.*)/,
+      'g'
+    );
+    const fileContents = readFileSync(configPath, { encoding: 'utf-8' });
+    if (!displayNameValueRegex.test(fileContents)) {
+      console.warn(stripIndents`Could not find "displayName" in ${configPath}. 
+      Please add a "displayName" to the Jest Config File.
+      Skipping this project (${task.split(':')[0]})...`);
+      continue;
+    }
+
+    const displayName = fileContents
+      .match(displayNameValueRegex)
+      .map((value) => value.substring(1, value.length - 1))[0];
+    selectedProjects.push(displayName);
```

Comment:
@barbados-clemens Here is the code to find the actual jest project 😄

Reply:
this should be fine for now, but there is a snowballs chance in hell that someone is using a var to reference the displayName. i.e. 

```ts
const myName = 'something-cool';

export default {
  displayName: myName
}
```
also more likely, 
```ts
export default {
  displayName: {
    name: 'something-cool',
    color: 'blue'
  }
}
```
in which case we'd error/warn about it missing when it was just configured differently. 

I think for an alpha it's fine, but we should make sure we handle this at some point. 

so maybe we should be more explicit in the error we are expecting the displayName to be a string for now?